### DISCUSSION TITLE
Web-UI behind apache reverse proxy

### DISCUSSION THREAD

**Discussion by jquiros2** [mrlt8/docker-wyze-bridge#744] [Q&A]
Hi! 
I have set all ports to only listen on 127.0.0.1 and cameras are accessible from the same host, where I am running zoneminder and have all feeds (monitors) set up ok.

I would like to secure the docker-wyze-bridge WebUI a bit so I am first trying to set up a reverse proxy to 127.0.0.1:5000 and once I get this working will move on to enabling apache's basic auth,

I have this section in one of my site configs (SSL enabled):

```
SSLProxyEngine on
ProxyRequests off
#ProxyPreserveHost On (on or off doesn't fix it)

<Location /wbr>
        ProxyPass http://127.0.0.1:5000
        ProxyPassReverse http://127.0.0.1:5000
        Order allow,deny
        Allow from all
</Location>
```
And it proxies almost nicely when adding the /wbr URL to that host's address, except I get no images.

I tried changing WB_IP a bit but no dice, and also tried setting "ProxyPreserveHost On" but can't get this scenario working yet.

Any pointers are very much appreciated!

**Comment by mrlt8** [comment]
Something like this?
```
ProxyPass /wbr http://127.0.0.1:5000/
ProxyPassReverse /wbr http://127.0.0.1:5000/
```

**Comment by jquiros2** [comment]
For some reason I couldn't get it to pass the /wbr part of the url (even as above) when it was in another config (same fqdn, but rying to separate target at "/wbr"

I was adding it into another host conf, but simplified by just creating another vhost.  Leaving this in case it's useful for ref for other ppl starting.  This will also block unauthenticated requests to /api.


This was first set up on port 80 without the ssl parts and just let "certbot --apache" create the cert and add the redirect rules.

```
<IfModule mod_ssl.c>
<VirtualHost *:443>
    ServerAdmin email@host.net
    ServerName yourfqdn
    DocumentRoot /var/www/html


SSLProxyEngine on
ProxyRequests off
ProxyPreserveHost on

<Location />
ProxyPass http://127.0.0.1:5000/
ProxyPassReverse http://127.0.0.1:5000/

        AuthType Basic
        AuthName "Restricted Content"
        AuthUserFile /yourpath/.htpassssfile
        Require valid-user
</Location>

    ErrorLog ${APACHE_LOG_DIR}/yourhost_error.log
    CustomLog ${APACHE_LOG_DIR}/yourhost_access.log combined

SSLCertificateFile /etc/letsencrypt/live/yourhost/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/yourhost/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
</IfModule>
```

Thanks @mrlt8 !  The bridge helps us get much more value from our several wyze cams, esp to tie into a multiuser environment like zm.  Having to depend on the wyze app sucks- anyone you share a cam with can power it off.
